### ISSUE TITLE
Barbarians - Issues

### ISSUE THREAD

**Issue by bob0bob** [ldtteam/minecolonies#1688]
minecolonies-universal-
minecraft 1.12.2-
mine colonies 0.8.4958

I'm experiencing several issues related to barbarians.

1.   Guard doesn't see raiders as a hostile mob.   I actual saw a guard walk right through barbarians go to their next waypoint on their patrol path.  Never reacted to the barbarians.   It appears their path is actually more important then a barbarians.  But  react as they should to a regular minecraft mob.

2.  Guards (my opinion) rush to protect citizens that are under attack, specially if they are not actively fighting anyone.

3.  Citizens reaction to raids are so wrong.   Had a raid and they didn't get they until morning while citizens were working.   It was a slaughter.  Have 12 citizens, when I got back into town to kill the barbarians.  There was 2 citizens left the guards.    The guards killed not one barbarians,  they where just patrolling like normal.   I saw 2 citizens as I was approaching that were not even running from the barbarians.   The barbarians where just standing in one place.   The 2 citizens appeared stuck in one tick to run away and would run away for a few block and then run back towards their working hut (like fish hut, farmer hut).  Then when they get a few blocks going the other way, turn back.   So they where just running back any forth.  Allowing barbarians to stand in one place hitting them as they run like a chicken with their heads cut off.
  Citizens should run back to their huts for safety or the town hall.

4. Guards didn't kill one barbarian,  I ended up killing them.

5.  You guys added a saying in the chat about who killing the guard and citizens.  That kill has issues.  when a guard killed the citizen, it just said the language text not the english text.   I don't remember it but it was the text for the guards name.  it was like  minecolonies.guard.name.   something like that appeared

**Comment by Raycoms** [comment]
@Asherslab

**Comment by bob0bob** [comment]
seeing now that barbarians spawn right at dusk.  Of course in the middle of town,  no father than 20 - 30 blocks from town hall.  This allows the barbarians to kill many citizens because most of them are not even in the houses yet.   Of course the citizens doesn't react to barbarians,   like such.   Run away faster.  Normal speed.

Also, I would think that once citizens are under attack that the guards would react but nothing from guards.

There are many holes in the barbarians code.   Of course walls mean nothing anymore since they spawn in the middle of town.  You colony has no hope of losing people no matter what you do,  with current coding that is all related to barbarian raiding.

Just wanted to give everything I'm seeing related to raids.

1.  Citizens do not react to barbarians being close. Citizens appear to do what is coded for time of day as the priority
2.  Guards do not react.  Guards have to run into barbarians and appears that have to finish getting to their way point if on patrol before reacting.
3.  barbarians spawn inside of colony. Should be on the outskirt of town, min.
4.  Guards do not see barbarians as a threat at first glance.  It takes many ticks before they react.  Even walking past them first. Guards should respond to citizens being attacked.
5.  Had a whole town of citizens killed without one guard attacking barbarians.   I didn't have guard guarding middle of town, until I figured out that guarding post on the edges of town is pointless because the threat is the middle of town from barbarians.
6.  The retreat on the guards (gui interface),  doesn't appear to be working well.  The barbarians walk at a normal pace and the retreat doesn't appear to have any more affect than a normal walk back to the tower.  Meaning barbarians keep up.  Typically a treat is at a fast pace. (my thought, this is up to the team).

These are my thoughts on the barbarians.   Since I finally figured out everything I could about barbarians.  I'm turning them off until the wonderful team and have time to deal with some of the issues about barbarians.

Keep up the great work.  Just can't wait for all the stuff I know you guys are planning.
